app/code/community/Twispay/Tpay/controllers/PaymentController.php

```diff
@@ -1,362 +1,280 @@
+        $apiKey = Mage::getStoreConfig('payment/tpay/stagingApiKey', $storeId);
+        $url = 'https://secure-stage.twispay.com';
+      }
+      Mage::Log(__FUNCTION__ . ': siteId=' . $siteId . ' apiKey=' . $apiKey . ' url=' . $url, Zend_Log::DEBUG, $this->logFileName);
+
+      /* Extract the billind and shipping addresses. */
+      $billingAddress = $order->getBillingAddress();
+      $shippingAddress = $order->getShippingAddress();
+
+      /* Extract the customer details. */
+      $customer = [ 'identifier' => (0 == $billingAddress->getCustomerId()) ? ('_' . $orderId . '_' . date('YmdHis')) : ('_' . $billingAddress->getCustomerId())
+                  , 'firstName' => ($billingAddress->getFirstname()) ? ($billingAddress->getFirstname()) : ($shippingAddress->getFirstname())
+                  , 'lastName' => ($billingAddress->getLastname()) ? ($billingAddress->getLastname()) : ($shippingAddress->getLastname())
+                  , 'country' => ($billingAddress->getCountryId()) ? ($billingAddress->getCountryId()) : ($shippingAddress->getCountryId())
+                  , 'state' => (('US' == $billingAddress->getCountryId()) && (NULL != $billingAddress->getRegionCode())) ? ($billingAddress->getRegionCode()) : ((('US' == $shippingAddress->getCountryId()) && (NULL != $shippingAddress->getRegionCode())) ? ($shippingAddress->getRegionCode()) : (''))
+                  , 'city' => ($billingAddress->getCity()) ? ($billingAddress->getCity()) : ($shippingAddress->getCity())
+                  , 'address' => ($billingAddress->getStreet()) ? ($billingAddress->getStreet()) : ($shippingAddress->getStreet())
+                  , 'zipCode' => ($billingAddress->getPostcode()) ? ($billingAddress->getPostcode()) : ($shippingAddress->getPostcode())
+                  , 'phone' => ($billingAddress->getTelephone()) ? ('+' . preg_replace('/([^0-9]*)+/', '', $billingAddress->getTelephone())) : (($shippingAddress->getTelephone()) ? ('+' . preg_replace('/([^0-9]*)+/', '', $shippingAddress->getTelephone())) : (''))
+                  , 'email' => ($billingAddress->getEmail()) ? ($billingAddress->getEmail()) : ($shippingAddress->getEmail())
+                  /* , 'tags' => [] */
+                  ];
+
+      /* Extract the items details. */
+      $items = array();
+      foreach($order->getAllItems() as $item){
+        $items[] = [ 'item' => $item->getName()
+                   , 'units' =>  (int) $item->getQtyOrdered()
+                   , 'unitPrice' => (string) number_format((float) $item->getPriceInclTax(), 2, '.', '')
+                   /* , 'type' => '' */
+                   /* , 'code' => '' */
+                   /* , 'vatPercent' => '' */
+                   /* , 'itemDescription' => '' */
+                   ];
+      }
 
-if (!function_exists('boolval')) {
-    function boolval($val) {
-            return (bool) $val;
+      /* Check if shiping price needs to be added. */
+      if(0 < $order->getShippingAmount()){
+        $index = count($items);
```

Comment:
The value of `$index` is not used anywhere (line 97 override any previous values)

Reply:
I removed the line.